src/engine/strat_engine/backstore/metadata/bda.rs

```diff
@@ -332,43 +381,40 @@ impl StaticHeader {
                 StaticHeader::sigblock_from_buf(&buf_loc_1),
                 StaticHeader::sigblock_from_buf(&buf_loc_2),
             ) {
-                (Ok(loc_1), Ok(loc_2)) => match (loc_1, loc_2) {
-                    (Some(loc_1), Some(loc_2)) => {
-                        if loc_1 == loc_2 {
-                            Ok(Some(loc_1))
-                        } else if loc_1.initialization_time == loc_2.initialization_time {
-                            // Inexplicable disagreement among static headers
-                            let err_str =
-                                "Appeared to be a Stratis device, but signature blocks disagree.";
-                            Err(StratisError::Engine(ErrorEnum::Invalid, err_str.into()))
-                        } else if loc_1.initialization_time > loc_2.initialization_time {
-                            // If the first header block is newer, overwrite second with
-                            // contents of first.
-                            BDA::write(f, &buf_loc_1, MetadataLocation::Second)?;
-                            Ok(Some(loc_1))
-                        } else {
-                            // The second header block must be newer, so overwrite first
-                            // with contents of second.
-                            BDA::write(f, &buf_loc_2, MetadataLocation::First)?;
-                            Ok(Some(loc_2))
+                (Ok(loc_1), Ok(loc_2)) => {
```

Comment:
Please drop the extra brace here, you lost that change on the rebase, and I'ld definitely like to keep it.